### PR TITLE
Ab#81933 document repository

### PR DESCRIPTION
**Pull request by TC-Cory**
Initial pattern for our document service repository including unit tests

### CODE REVIEW

File: DocumentService/Repositories/DocumentRepository.cs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            {
+                throw;
+            }
+
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<DocumentInfo> GetDocumentsByIds(Guid[] ids)
+        {
+            return this.Filter(x => ids.Contains(x.DocumentId));
```
Comment (tc-chris) [inline DocumentService/Repositories/DocumentRepository.cs:125]: ids.Equals() might be best

File: DocumentService/Repositories/DocumentRepository.cs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        }
+
+        /// <summary>
+        /// Function to return a single document that is not set for deletion
+        /// </summary>
+        /// <param name="id">Id of document to find</param>
+        /// <returns>Document found</returns>
+        private async Task<DocumentInfo> GetDocument(Guid id)
+        {
+            var query = from q in context.DocumentInfo
```
Comment (tc-chris) [inline DocumentService/Repositories/DocumentRepository.cs:141]: Can we change this to Linq as it more clean

File: DocumentService/Startup.cs [hunk 3]
```diff
@@ -30,8 +23,7 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddDbContext<DocumentContext>(options => options.UseNpgsql(Configuration.GetConnectionString("DocumentContext")));
-            services.AddSingleton<IDocumentsInitializer, DocumentsInitializer>();
-          
+            services.AddScoped<IDocumentRepository, DocumentRepository>();
```
Comment (tc-chris) [inline DocumentService/Startup.cs:26]: Do we not want this to be a singleton? As it would only open the connection once

**Review by tc-chris — CHANGES_REQUESTED** [review]

File: DocumentService/Models/DocumentInfo.cs [hunk 4]
```diff
@@ -114,6 +115,35 @@ public class DocumentInfo
...
+        public virtual Correlation Correlation { get; set; }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as DocumentInfo);
+        }
+
+        public bool Equals(DocumentInfo other)
+        {
+            return other != null &&
```
[the remark this answers is not in the document]
Comment (AlexandreSanscartier) [reply DocumentService/Models/DocumentInfo.cs:128]: I tried the tests out of curiosity and they worked anyway so I think we can just remove this.

File: DocumentService/Repositories/DocumentRepository.cs (hunk 1 above)
Comment (AlexandreSanscartier) [reply DocumentService/Repositories/DocumentRepository.cs:125]: here we're checking if the list contains the DocumentId, Equals would not work in our case.

File: DocumentService/Repositories/DocumentRepository.cs (hunk 2 above)
Comment (AlexandreSanscartier) [reply DocumentService/Repositories/DocumentRepository.cs]: Done.

File: DocumentService/Models/DocumentTypes.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DocumentService.Models
+{
+    public class DocumentTypes
+    {
+        public int DocumentTypesId { get; set; }
+        public string DocType { get; set; }
```
[the remark this answers is not in the document]
Comment (AlexandreSanscartier) [reply DocumentService/Models/DocumentTypes.cs]: Done.

File: DocumentService/Startup.cs (hunk 3 above)
Comment (tc-chris) [reply DocumentService/Startup.cs:26]: Okay, that makes sense

File: DocumentService/Repositories/DocumentRepository.cs (hunk 1 above)
Comment (tc-chris) [reply DocumentService/Repositories/DocumentRepository.cs:125]: Sounds good

**Pull request merged**